eis_toolkit/prediction/mlp.py

```diff
@@ -0,0 +1,301 @@
+from numbers import Number
+from typing import Literal, Optional, Sequence, Tuple
+
+import numpy as np
+from beartype import beartype
+from keras.optimizers import SGD, Adagrad, Adam, RMSprop
+from tensorflow import keras
+
+from eis_toolkit import exceptions
+
+
+def _keras_optimizer(optimizer: str, **kwargs):
+    """Create a Keras optimizer from given name and parameters."""
+    if optimizer == "adam":
+        return Adam(**kwargs)
+    elif optimizer == "adagrad":
+        return Adagrad(**kwargs)
+    elif optimizer == "rmsprop":
+        return RMSprop(**kwargs)
+    elif optimizer == "sdg":
+        return SGD(**kwargs)
+    else:
+        raise exceptions.InvalidParameterValueException(f"Unidentified optimizer: {optimizer}")
+
+
+def _check_MLP_inputs(
+    neurons: Sequence[int],
+    validation_split: Optional[float],
+    learning_rate: float,
+    dropout_rate: Optional[Number],
+    es_patience: int,
+    batch_size: int,
+    epochs: int,
+    output_neurons: int,
+    loss_function: str,
+) -> None:
+    """Check parameters for Keras MLP training."""
+    if len(neurons) == 0:
+        raise exceptions.InvalidParameterValueException("Neurons parameter must be a non-empty list.")
+
+    if any(neuron < 1 for neuron in neurons):
+        raise exceptions.InvalidParameterValueException("Each neuron in neurons list must be at least 1.")
+
+    if validation_split and not (0 < validation_split < 1):
+        raise exceptions.InvalidParameterValueException("Validation split must be a value between 0 and 1, exclusive.")
+
+    if learning_rate <= 0:
+        raise exceptions.InvalidParameterValueException("Learning rate must be greater than 0.")
+
+    if dropout_rate and not (0 <= dropout_rate <= 1):
+        raise exceptions.InvalidParameterValueException("Dropout rate must be between 0 and 1, inclusive.")
+
+    if es_patience <= 0:
+        raise exceptions.InvalidParameterValueException("Early stopping patience must be greater than 0.")
+
+    if batch_size <= 0:
+        raise exceptions.InvalidParameterValueException("Batch size must be greater than 0.")
+
+    if epochs <= 0:
+        raise exceptions.InvalidParameterValueException("Number of epochs must be greater than 0.")
+
+    if output_neurons <= 0:
+        raise exceptions.InvalidParameterValueException("Number of output neurons must be greater than 0.")
+
+    if output_neurons > 1 and loss_function == "binary_crossentropy":
```

Comment:
Yes, good idea. But is `output_neurons` == 2 then invalid in all cases, or am I missing something?

Reply:
Since you included `sparse_categorical_crossentropy` in this check, I assumed it should also be added as a parameter value for `loss_function` in `train_MLP_classifier` ;). Let me know if this wasn't the idea

Reply:
Okay :+1: . Again I would let you decide whether to add `sparse_categorical_crossentropy` or not. We have one-hot enconding available for users and can even perform it automatically (inside MLP or inside CLI for plugin users), if we want. I would just add to the docstring that we expect one-hot encoded data if that is the case, or that it depends on the loss function if that is the case.